xarray/backends/common.py

```diff
@@ -535,6 +535,17 @@ def open_datatree(
 
         raise NotImplementedError()
 
+    def open_groups(
```

Comment:
```suggestion
    def open_groups_as_dict(
```
This was @shoyer's suggestion - keep the top-level public-facing function called `open_groups`, but change the name of this method to help avoid confusion with the `open_group` method defined on the zarr backend.